pxr/base/tf/stringUtils.h

```diff
@@ -683,18 +727,113 @@ std::string TfStringCatPaths( const std::string &prefix,
+    {
+        // use unicode utils to validate the identifier
+        return TfUnicodeUtils::IsValidUTF8Identifier(identifier, 
+            identifier.begin(), identifier.end());
+    }
+    else
+    {
+        char const *p = identifier.c_str();
+        auto letter = [](unsigned c) { return ((c-'A') < 26) || ((c-'a') < 26); };
+        auto number = [](unsigned c) { return (c-'0') < 10; };
+        auto under = [](unsigned c) { return c == '_'; };
+        unsigned x = *p;
+        if (!x || number(x)) {
+            return false;
+        }
+        while (letter(x) || number(x) || under(x)) {
+            x = *p++;
+        };
+        return x == 0;
     }
-    while (letter(x) || number(x) || under(x)) {
-        x = *p++;
-    };
-    return x == 0;
+}
+
+/// Test whether the subsequence of \a identifier given by begin and end 
+/// is a valid identifier name.
+///
+/// An identifier name is valid if it consists of any valid UTF-8 sequence 
+/// that is at least one character long and fulfills the standard C/Python 
+/// identifier convention for Unicode (XID_Start followed by 0 or more 
+/// XID_Continue class characters).  If the TF_UTF8_IDENTIFIERS environment 
+/// setting is overridden to false, identifier validity is checked against 
+/// legacy rules and follows the C/Python identifier convention for ASCII 
+/// characters (i.e. at least one character long, must start with a letter 
+/// or underscore, and must contain only letters, underscores, and numerals).
+/// WARNING: This method is for internal use only and performs no 
+///          validation on begin / end.
+/// 
+inline bool
+_TfIsValidIdentifierSubsequence(const std::string& identifier,
+    std::string::const_iterator begin, std::string::const_iterator end)
+{
+    if (UseUTF8Identifiers())
+    {
+        // use unicode utils to validate the identifier
+        return TfUnicodeUtils::IsValidUTF8Identifier(identifier, 
+            begin, end);
+    }
+    else
+    {
+        size_t count = (end - begin);
+        char const *p = &(*begin);
+        auto letter = [](unsigned c) { return ((c-'A') < 26) || ((c-'a') < 26); };
+        auto number = [](unsigned c) { return (c-'0') < 10; };
+        auto under = [](unsigned c) { return c == '_'; };
+        unsigned x = *p;
+        if (!x || number(x)) {
+            return false;
+        }
+        for (size_t i = 0; i < count; i++)
+        {
+            if (letter(x) || number(x) || under(x))
+            {
+                x = *p++;
+            }
+            else
+            {
+                break;
+            }
+        }
+        return x == 0;
+    }
+}
+
+/// Tests whether \a primName is valid.
+///
+/// A prim name is valid if it's at least one character long and all characters
+/// are valid UTF-8 sequences.  This method ignores the value of 
+/// TF_UTF8_IDENTIFIERS and instead uses either UTF-8 or ASCII rules for 
+/// validation based on the value of \a utf8.
+///
+inline bool
```

Comment:
I was under the impression that we were going to expand prim & property names to be UTF-8 style identifiers (xid_start + xid_continue), not any arbitrary sequence of utf-8 encoded code points.  That seems a bit bananas to me, a single space being a valid prim name for example?  Or allowing control codes, or SdfPath syntax characters?

Reply:
Yes, this comment leaves out the part below "Valid UTF-8 sequences here means all characters must be in the XID_Continue character class".  This additional comment should be there.

Reply:
Added this comment when lofting those methods up to Sdf